File: src/main/java/me/lovesasuna/bot/function/colorphoto/pixiv.java
package me.lovesasuna.bot.function.colorphoto;

import com.fasterxml.jackson.databind.JsonNode;
import kotlin.Pair;
import me.lovesasuna.bot.Main;
import me.lovesasuna.bot.data.BotData;
import me.lovesasuna.bot.util.NetWorkUtil;

import java.io.IOException;
import java.io.InputStream;

/**
 * @author LovesAsuna
 * @date 2020/4/19 14:06
 */
public class pixiv implements Source{
    @Override
    public String fetchData() {
        String source = "https://api.lolicon.app/setu/?apikey=560424975e992113ed5977";
        Pair<InputStream,Integer> result = NetWorkUtil.fetch(source);
        try {
            InputStream inputStream = result.getFirst();
            JsonNode root = BotData.INSTANCE.getObjectMapper().readTree(inputStream);
            return root.get("data").get(0).get("url").asText() + "|" + root.get("quota").asText();
        } catch (IOException | NullPointerException e) {
            e.printStackTrace();
            return null;
        }
    }
}

File: src/main/java/me/lovesasuna/bot/function/colorphoto/Pixiv.kt
package me.lovesasuna.bot.function.colorphoto

import me.lovesasuna.bot.data.BotData.objectMapper
import me.lovesasuna.bot.data.pushError
import me.lovesasuna.bot.util.interfaces.PhotoSource
import me.lovesasuna.lanzou.util.NetWorkUtil
import java.io.IOException

/**
 * @author LovesAsuna
 * @date 2020/4/19 14:06
 */
class Pixiv : PhotoSource {
    override fun fetchData(): String? {
        // 备用976835505edf70ff564238
        val source = "https://api.lolicon.app/setu/?apikey=560424975e992113ed5977"
        val result = NetWorkUtil[source]
        return try {
            val inputStream = result!!.second
            val root = objectMapper!!.readTree(inputStream)
            val quota = root["quota"].asText()
            val url = root["data"][0]?.let { it["url"].asText() } ?: return "|0"
            return "$url|$quota"
        } catch (e: IOException) {
            e.pushError()
            e.printStackTrace()
            null
        } catch (e: NullPointerException) {
            e.pushError()
            e.printStackTrace()
            null
        }
    }
}